test/test_package.cpp

```diff
@@ -1804,3 +1794,29 @@ TEST(DDPackageTest, InnerProductTopNodeConjugation) {
     // it will result in +0.416.
     EXPECT_NEAR(dd->expectationValue(op, evolvedState), -0.416, 0.001);
 }
+
+TEST(DDPackageTest, QubitGateIdentityRemoval) {
+    auto dd        = std::make_unique<dd::Package<>>(5);
+    auto hGate     = dd->makeGateDD(dd::Hmat, 5, 0);
+    auto zeroState = dd->makeZeroState(5);
+    EXPECT_EQ(dd->size(hGate), 2);
+
+    auto newState = dd->multiply(hGate, zeroState);
+    dd->printVector(newState);
+
+    ASSERT_EQ(dd->getValueByPath(newState, 0), (dd::ComplexValue{dd::SQRT2_2, 0}));
+    ASSERT_EQ(dd->getValueByPath(newState, 1), (dd::ComplexValue{dd::SQRT2_2, 0}));
+    ASSERT_EQ(dd->getValueByPath(newState, 2), (dd::ComplexValue{0, 0}));
+    ASSERT_EQ(dd->getValueByPath(newState, 3), (dd::ComplexValue{0, 0}));
+
+    auto cxGate = dd->makeGateDD(dd::Xmat, 5, 0_pc, 1);
+    EXPECT_EQ(dd->size(cxGate), 4);
```

Comment:
```suggestion
    EXPECT_EQ(dd->size(cxGate), 3);
```

Something I missed in the review. The CX gate should just require two nodes (plus the terminal).

Side note: we should probably adjust the size function so that it really only counts non-terminal nodes. This would be a breaking change though and should be done in a separate PR at some point.